mDNSWindows/mDNSWin32.c

```diff
@@ -1947,9 +1947,7 @@ SetDomainFromDHCP( void )
 
 	for ( pAdapter = pAdapterInfo; pAdapter; pAdapter = pAdapter->Next )
 	{
-		if ( pAdapter->IpAddressList.IpAddress.String &&
```

Comment:
What's the warning here? Why would we not want to check the pointer for NULL?

Reply:
It's not a pointer, but a [C array](https://docs.microsoft.com/en-us/windows/win32/api/iptypes/ns-iptypes-ip_address_string).

`warning: address of array 'pAdapter->IpAddressList.IpAddress.String' will always evaluate to 'true' [-Wpointer-bool-conversion]`